### ISSUE TITLE
iOS 11 - Bug with shareViaTwitter, empty text field on share sheet

### ISSUE THREAD

**Issue by hellpoke** [EddyVerbruggen/SocialSharing-PhoneGap-Plugin#879]
Hi,

Since iOS 11, I've a problem with the function window.plugins.socialsharing.shareViaTwitter() like this : 
`window.plugins.socialsharing.shareViaTwitter(msg, picture, link)`

So I can share my card, my picture appear in the share sheet but the text field is filling by default text twitter "What's happening?". Screenshot : 

![image](https://user-images.githubusercontent.com/27951547/30975181-e874d118-a471-11e7-83e3-88a341091d1c.png)

I checked if my var msg wasn't empty but that good. When I exchange picture by null, msg is load and replace default text but, of course, my picture doesn't appear.

Anyone can explain me what's happen?
Thank you,

**Comment by andrzejtemplin** [comment]
I confirm that bug. It doesn't appear on Android though. As a temporary workaround I copy text to the clipboard by myself and handle this in similar manner as shareWithFacebookWithHint.

**Comment by a930334** [comment]
I confirm that bug, but only on iOS. On Android works Fine.
I have no problem if i use "shareWithOptions", but the problem appears if i use "shareViaTwitter"...

Regards

**Comment by a930334** [comment]
I'm using cordova-plugin-x-socialsharing 5.2.1

**Comment by stewwan** [comment]
+1 =(

**Comment by stewwan** [comment]
@EddyVerbruggen any idea?

**Comment by EddyVerbruggen** [comment]
No, but [Google found this](https://twittercommunity.com/t/bug-in-ios-11/91490/28)

**Comment by stewwan** [comment]
thanks for replying @EddyVerbruggen 

it seems now have to use the TWTRComposer instead of SLComposeViewController

https://stackoverflow.com/questions/47218557/cannot-share-both-image-and-text-on-twitter-using-slcomposeviewcontroller
